src/test/java/th/ac/kmitl/science/comsci/example/nameprinters/NamePrinterLoaderTest.java

```diff
@@ -0,0 +1,18 @@
+package th.ac.kmitl.science.comsci.example.nameprinters;
+
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+public class NamePrinterLoaderTest {
+    @Test
+    public void getNamePrinters() {
+        int namePrinterCount = 0;
+
+        for(NamePrinter namePrinter : NamePrinterLoader.getInstance())
+            namePrinterCount++;
+
+        assertEquals(namePrinterCount > 0, true);
```

Comment:
you should test that list size is correct and data in list is correct instead of "> 0"

Reply:
@champillon Is there anyway to get a correct size of the list to assert or I have to hard-coded in this test?

Reply:
If it is test, you need to mock it.

if it work correctly for 1, it will work correctly for n.

Reply:
@champillon I added some setters to allow mocking list of `NamePrinter` classes of `NamePrinterLoader`, the mock setup method is inside this test class on current snapshot.